combat-trainer.lic

```diff
@@ -4000,8 +4019,14 @@ class GameState
     true
   end
 
-  def update_analyze_array(only_accuracy)
-    return perform_analyze?('accuracy', 50) if only_accuracy # Barb only setting
+  def update_analyze_array(type = '')
+    # Force overrides during whirlwinding, barb only
+    if currently_whirlwinding
+      return perform_analyze?('accuracy', 50) if type == 'accuracy'
+      return perform_analyze?('flame', 0) if type == 'flame'
+    end
```

Comment:
Now that we've updated the conditionals that may call `update_analyze_array` with either `flame` or `accuracy` arguments, is the `if currently_whirlwinding` check here still necessary?

Maybe we need a way at line 4032 to go to the next item in the loop until the "type" key matches the "type" method argument?

Something akin to:
```ruby
    if DRStats.barbarian?
      { 'accuracy' => 50, 'damage' => 125, 'balance' => 600, 'flame' => 0 }.each do |combo, expertise|
        if type.nil? || type.empty? || type == combo
          break if perform_analyze?(combo, expertise)
        end
      end
    else
      perform_analyze?('', 0)
    end
```

Reply:
Actually, I think a better approach is more like what you had with a minor modification to not be whirlwind specific.

```ruby
  def update_analyze_array(type = nil)
    if DRStats.barbarian?
      analyze_map = { 'accuracy' => 50, 'damage' => 125, 'balance' => 600, 'flame' => 0 }
      return if type && perform_analyze?(type, analyze_map[type])
      analyze_map.each do |combo, expertise|
        break if perform_analyze?(combo, expertise)
      end
    else
      perform_analyze?('', 0)
    end
  end
```

Reply:
Updated it with this, and moved map to class variable.

Reply:
Thanks!